File: swift sms gateway /get_balance.swift
import Foundation

let appID = "Api Key"
let appSecret = "Api Secret"
let appHash = Data("\(appID):\(appSecret)".utf8).base64EncodedString()
let baseURL = "https://api-sms.4jawaly.com/api/v1/"

let query: [String: String] = [:] // Define the query parameters here if needed
let queryString = query.map { "\($0.key)=\($0.value)" }.joined(separator: "&")
let urlString = baseURL + "account/area/me/packages?" + queryString

guard let url = URL(string: urlString) else {
    print("Invalid URL")
    exit(1)
}

var request = URLRequest(url: url)
request.httpMethod = "GET"
request.setValue("application/json", forHTTPHeaderField: "Accept")
request.setValue("application/json", forHTTPHeaderField: "Content-Type")
request.setValue("Basic \(appHash)", forHTTPHeaderField: "Authorization")

let task = URLSession.shared.dataTask(with: request) { data, response, error in
    guard let data = data, let response = response as? HTTPURLResponse else {
        print("Error: \(error?.localizedDescription ?? "Unknown error")")
        return
    }

    if response.statusCode != 200 {
        print("Error code: \(response.statusCode)")
        exit(0)
    }

    do {
        if let responseJSON = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
            if let errorCode = responseJSON["code"] as? Int, errorCode == 200 {
                if let totalBalance = responseJSON["total_balance"] as? Double {
                    print("Your balance: \(totalBalance)")
                }
            } else {
                if let message = responseJSON["message"] as? String {
                    print(message)
                }
            }
        }
    } catch {
        print("Error parsing JSON: \(error.localizedDescription)")
    }
}

task.resume()

File: php/Get_balance.php
<?php

$app_id = "API_KEY";     // Replace with your actual API KEY
$app_sec = "API_SECRET"; // Replace with your actual API SECRET

$app_hash = base64_encode("$app_id:$app_sec");
$base_url = "https://api-sms.4jawaly.com/api/v1/";

// Build query params
$query = [];
$query["is_active"] = 1; // get active only
$query["order_by"] = "id"; // or package_points, current_points, expire_at
$query["order_by_type"] = "desc"; // desc or asc
$query["page"] = 1;
$query["page_size"] = 10;
$query["return_collection"] = 1; // get all collection

$url = $base_url . 'account/area/me/packages?' . http_build_query($query);

$headers = [
    'Accept: application/json',
    'Content-Type: application/json',
    'Authorization: Basic ' . $app_hash
];

// Initialize curl
$curl = curl_init();
curl_setopt_array($curl, array(
    CURLOPT_URL => $url,
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_ENCODING => '',
    CURLOPT_MAXREDIRS => 10,
    CURLOPT_TIMEOUT => 0,
    CURLOPT_FOLLOWLOCATION => true,
    CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
    CURLOPT_CUSTOMREQUEST => 'GET',
    CURLOPT_HTTPHEADER => $headers,
));

$response = curl_exec($curl);

if (curl_errno($curl)) {
    die("cURL Error: " . curl_error($curl));
}
$http_code = curl_getinfo($curl, CURLINFO_HTTP_CODE);

curl_close($curl);

// Parse response
$data = json_decode($response, true);

header('Content-Type: application/json; charset=utf-8');

// Handle different outcomes
if ($http_code == 200 && isset($data["code"]) && $data["code"] == 200) {
    if (isset($data["collection"]) && !empty($data["collection"])) {
        // Print full collection as pretty JSON
        echo json_encode($data["collection"], JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
    } elseif (isset($data["total_balance"])) {
        // Print only total balance
        echo json_encode(["total_balance" => $data["total_balance"]], JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
    } else {
        // Unexpected success response
        echo json_encode(["message" => "No package data found.", "response" => $data], JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
    }
} else {
    // API error
    $error_msg = $data["message"] ?? "Failed to fetch packages data";
    echo json_encode(["message" => $error_msg, "error" => $data], JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
}
